client/src/app.tsx

```diff
@@ -130,18 +133,36 @@ class App extends React.Component<IAppProps, IAppState> {
                         <InfoPanel/>
                     </div>
                 )}
-                <div className="actions-panel-outer floating-panel">
-                    <div className="actions-panel-inner">
-                        <ActionsPanel
-                            orbitDistance={this.orbitDistanceSubject}
-                            islandState={this.state.islandState}
-                            location={this.perspectiveCamera.position}
-                        />
+                {this.state.islandState ? (
+                    <div className="actions-panel-outer floating-panel">
+                        <div className="actions-panel-inner">
+                            <ActionsPanel
+                                orbitDistance={this.orbitDistanceSubject}
+                                islandState={this.state.islandState}
+                                location={this.perspectiveCamera.position}
+                            />
+                        </div>
                     </div>
-                </div>
+                ) : (
+                    <h1>No island!</h1>
+                )}
             </div>
         )
     }
+
+    private fetchIsland(islandName: string): void {
+        this.props.storage.getIslandData(islandName).then(islandData => {
```

Comment:
This would read nicer if you used async/await instead https://javascript.info/async-await

Reply:
ooh cool yeah that's the thing i wanted to learn